__tests__/app.test.js

```diff
@@ -69,48 +68,168 @@ describe("GET /api/articles", () => {
+  });
+  test("400: Returns with invalid article request if given an invalid data request (ie: not a number)", () => {
     return request(app)
-    .get("/api/articles/invalid_article_id")
-    .expect(400)
-    .then((response) => {
-      expect(response.body.msg).toBe("Invalid Request")
-    })
-  })
+      .get("/api/articles/invalid_article_id")
+      .expect(400)
+      .then(({body}) => {
+        expect(body.msg).toBe("Invalid Request");
+      });
+  });
   test("404: Returns with article ID not found if given a request of the correct data type that does not exist", () => {
     return request(app)
-    .get("/api/articles/3000")
-    .expect(404)
-    .then((response) =>{
-      expect(response.body.msg).toBe("Article ID Not Found")
+      .get("/api/articles/3000")
+      .expect(404)
+      .then(({body}) => {
+        expect(body.msg).toBe("Article ID Not Found");
+      });
+  });
+});
+
+describe("POST: /api/articles/:article_id/comments", () => {
+  test("201:Responds with newly created comment object", () => {
+    const newComment = {
+      body: "this is the new body",
+      username: "butter_bridge",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body: { comment } }) => {
+        comment.forEach((key) => {
+          expect(key).toHaveProperty("comment_id", expect.any(Number));
+          expect(key).toHaveProperty("body", expect.any(String));
+          expect(key).toHaveProperty("article_id", expect.any(Number));
+          expect(key).toHaveProperty("author", expect.any(String));
+          expect(key).toHaveProperty("votes", expect.any(Number));
+          expect(key).toHaveProperty("created_at", expect.any(String));
+        });
+      });
+  });
+
+  test("400: Returns with invalid request if given an invalid data request (ie: not a number)", () => {
+    const newComment = {
+      body: "this is the new body",
+      username: "butter_bridge",
+    };
+    return request(app)
+      .post("/api/articles/invalid_article_id/comments")
+      .send(newComment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Invalid Request");
+      });
+  });
+
+  test("404: Returns with article ID not found if given a request of the correct data type that does not exist", () => {
+    const newComment = {
+      body: "this is the new body",
+      username: "butter_bridge",
+    };
+    return request(app)
+      .post("/api/articles/3000")
+      .send(newComment)
+      .expect(404)
+      .then(({body}) => {
+        expect(body.msg).toBe("Path Not Found");
+      });
+  });
+  test("400: respond with bad request if post body contains missing field", () => {
+    const newComment = {
+      body: "this is the new body",
+    };
+    return request(app).post("/api/articles/1/comments")
+    .send(newComment)
+    .expect(400).then(({body}) => {
+      expect(body.msg).toBe("Missing Fields")
```

Comment:
Awesome tests for the errors, just one missing: 404 - valid body but the username is not found in the database! 👍